tiatoolbox/tools/patchextraction.py

```diff
@@ -87,14 +107,14 @@ class PatchExtractor(ABC):
 
     def __init__(
         self,
-        input_img,
-        patch_size,
-        input_mask=None,
-        resolution=0,
-        units="level",
-        pad_mode="constant",
-        pad_constant_values=0,
-        within_bound=False,
+        input_img: Union[str, Path, np.ndarray],
+        patch_size: Union[int, Tuple[int, int]],
+        input_mask: Union[str, Path, np.ndarray, WSIReader] = None,
+        resolution: Union[int, float, Tuple[float, float]] = 0,
+        units: str = "level",
+        pad_mode: str = "constant",
+        pad_constant_values: Union[int, Tuple[int, int]] = 0,
+        within_bound: bool = False,
```

Comment:
Just be mindful of adding the new argument `mask_ratio` that is being introduced in the latest PR #451 if that one is going to be merged first.

Reply:
I have done this when I merged develop into Roger's branch.